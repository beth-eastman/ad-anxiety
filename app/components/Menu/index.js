/* eslint-disable */
import React from 'react';
import MaterialMenu from 'material-ui/Menu';
import Divider from 'material-ui/Divider';
import MenuItem from 'material-ui/MenuItem';
import Popover from 'material-ui/Popover';
import VideoCam from 'material-ui/svg-icons/av/videocam';
import HomeIcon from 'material-ui/svg-icons/action/home';
import Assessment from 'material-ui/svg-icons/action/assessment';
import IconButton from 'material-ui/IconButton';
import LibraryBooks from 'material-ui/svg-icons/av/library-books';
import Resources from 'material-ui/svg-icons/action/info';
import AppHub from 'material-ui/svg-icons/action/exit-to-app';
import Hamburger from 'material-ui/svg-icons/navigation/menu';

export default class Menu extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      open: false,
    };
  }

  /* Open Menu when user selects menu icon*/
  handleTouchTap = (event) => {
    // This prevents ghost click.
    event.preventDefault();

    this.setState({
      open: true,
      anchorEl: event.currentTarget,
    });
  };

  /* Close the menu */
  handleRequestClose = () => {
    this.setState({
      open: false,
    });
  };

  render() {
    return (
      <div className="menu-popovers">
        <IconButton
          onTouchTap={this.handleTouchTap}
          iconStyle={{ fill: 'white' }}
        >
          <Hamburger />
        </IconButton>

        <Popover
          open={this.state.open}
          anchorEl={this.state.anchorEl}
          anchorOrigin={{ horizontal: 'left', vertical: 'bottom' }}
          targetOrigin={{ horizontal: 'left', vertical: 'top' }}
          onRequestClose={this.handleRequestClose}
        >
          <MaterialMenu>
            <MenuItem
              key={'home'}
              primaryText="Home"
              leftIcon={<HomeIcon />}
              onTouchTap={() => {
                this.props.router.push('/');
                this.handleRequestClose();
              }}
            />
            <Divider />
            <MenuItem
              key={'videos'}
              primaryText="Videos"
              leftIcon={<VideoCam />}
              onTouchTap={() => {
                this.props.router.push('/videos');
                this.handleRequestClose();
              }}
            />
            <MenuItem
              key={'assessments'}
              primaryText="Assessments"
              leftIcon={<Assessment />}
              onTouchTap={() => {
                this.props.router.push('/assessment');
                this.handleRequestClose();
              }}
            />
            <MenuItem
              key={'library'}
              primaryText="Library"
              leftIcon={<LibraryBooks />}
              onTouchTap={() => {
                this.props.router.push('/library');
                this.handleRequestClose();
              }}
            />
            <MenuItem
              key={'resources'}
              primaryText="Resources"
              leftIcon={<Resources />}
              onTouchTap={() => {
                this.props.router.push('/resources');
                this.handleRequestClose();
              }}
            />
            <Divider />
            <a
              href="https://apphub.tee2.org/#/"
              style={{ textDecoration: 'none' }}
            >
              <MenuItem
                key={'hub'}
                primaryText="App Hub"
                leftIcon={<AppHub />}
                onTouchTap={() => {
                  this.handleRequestClose();
                }}
              />
            </a>
          </MaterialMenu>
        </Popover>
      </div>
    );
  }
}
