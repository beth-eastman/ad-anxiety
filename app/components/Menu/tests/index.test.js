/* eslint-disable */
/*
 * index.test.js - Menu
 * Tests for the Menu components
 *
 * Created by Bethany Eastman on 06/21/17.
 *
 * AD Anxiety
 *
 * Copyright Â© 2009-2017 United States Government as represented by
 * the Chief Information Officer of the National Center for Telehealth
 * and Technology. All Rights Reserved.
 *
 * Copyright Â© 2009-2017 Contributors. All Rights Reserved.
 *
 * THIS OPEN SOURCE AGREEMENT ("AGREEMENT") DEFINES THE RIGHTS OF USE,
 * REPRODUCTION, DISTRIBUTION, MODIFICATION AND REDISTRIBUTION OF CERTAIN
 * COMPUTER SOFTWARE ORIGINALLY RELEASED BY THE UNITED STATES GOVERNMENT
 * AS REPRESENTED BY THE GOVERNMENT AGENCY LISTED BELOW ("GOVERNMENT AGENCY").
 * THE UNITED STATES GOVERNMENT, AS REPRESENTED BY GOVERNMENT AGENCY, IS AN
 * INTENDED THIRD-PARTY BENEFICIARY OF ALL SUBSEQUENT DISTRIBUTIONS OR
 * REDISTRIBUTIONS OF THE SUBJECT SOFTWARE. ANYONE WHO USES, REPRODUCES,
 * DISTRIBUTES, MODIFIES OR REDISTRIBUTES THE SUBJECT SOFTWARE, AS DEFINED
 * HEREIN, OR ANY PART THEREOF, IS, BY THAT ACTION, ACCEPTING IN FULL THE
 * RESPONSIBILITIES AND OBLIGATIONS CONTAINED IN THIS AGREEMENT.
 *
 * Government Agency: The National Center for Telehealth and Technology
 * Government Agency Original Software Designation: ProductName001
 * Government Agency Original Software Title: ProductName
 * User Registration Requested. Please send email
 * with your contact information to: robert.a.kayl.civ@mail.mil
 * Government Agency Point of Contact for Original Software: robert.a.kayl.civ@mail.mil
 *
*/
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { shallow, mount, render } from 'enzyme';
import Menu from '../index';

it('renders Menu', () => {
  const tree = renderer.create(
    <MuiThemeProvider>
      <Menu setPageTitle={function () {}} changeIcon={function () {}} />
    </MuiThemeProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

// it('opens and closes', () => {
//   let tree = renderer.create(
//     <MuiThemeProvider>
//       <Menu setPageTitle={function () {}} changeIcon={function () {}} />
//     </MuiThemeProvider>
//   );
//   // checkbox.find('IconButton').simulate('change');
//   // checkbox.find('IconButton').simulate('change');
//   // tree = tree.toJSON();
//
//   expect(tree.state('open')).to.equal('close');
//   // expect(tree).toMatchSnapshot();
// });

// it('opens and closes', () => {
//   const wrapper = mount(
//     <MuiThemeProvider>
//       <Menu setPageTitle={function () {}} changeIcon={function () {}} />
//     </MuiThemeProvider>
//   );
//   expect(wrapper.state().open).to.equal(false);
// });
