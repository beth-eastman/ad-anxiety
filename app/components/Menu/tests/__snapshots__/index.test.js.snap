exports[`test renders Menu 1`] = `
<div
  className="menu-popovers">
  <button
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": 48,
        "zIndex": 1,
      }
    }
    tabIndex={0}
    type="button">
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}>
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "white",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24">
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
      </svg>
    </div>
  </button>
  <div
    style={
      Object {
        "display": "none",
      }
    } />
</div>
`;
